FacebookClient.cs

```diff
@@ -7,51 +7,55 @@
-    {
-        Task<T> GetAsync<T>(string accessToken, string endpoint, string args = null);
-        Task PostAsync(string accessToken, string endpoint, object data, string args = null);
-    }
-
-    public class FacebookClient : IFacebookClient
-    {
-        private readonly HttpClient _httpClient;
-
-        public FacebookClient()
-        {
-            _httpClient = new HttpClient
-            {
-                BaseAddress = new Uri("https://graph.facebook.com/v10.0/")
-            };
-            _httpClient.DefaultRequestHeaders
-                .Accept
-                .Add(new MediaTypeWithQualityHeaderValue("application/json"));
-        }
-
-        public async Task<T> GetAsync<T>(string accessToken, string endpoint, string args = null)
-        {
-            var response = await _httpClient.GetAsync($"{endpoint}?access_token={accessToken}&{args}");
-            if (!response.IsSuccessStatusCode)
-            {
-                throw new Exception($"{(int)response.StatusCode} {response.ReasonPhrase} {await response.Content.ReadAsStringAsync()}");
-            }
-
-            var result = await response.Content.ReadAsStringAsync();
-
-            return JsonConvert.DeserializeObject<T>(result);
-        }
-
-        public async Task PostAsync(string accessToken, string endpoint, object data, string args = null)
-        {
-            var payload = GetPayload(data);
-            await _httpClient.PostAsync($"{endpoint}?access_token={accessToken}&{args}", payload);
-        }
-
-        private static StringContent GetPayload(object data)
-        {
-            var json = JsonConvert.SerializeObject(data);
-
-            return new StringContent(json, Encoding.UTF8, "application/json");
-        }
-    }
+	public interface IFacebookClient
+	{
+		Task<T> GetAsync<T>(string accessToken, string endpoint, string args = null);
+		Task PostAsync(string accessToken, string endpoint, object data, string args = null);
+	}
+
+	public class FacebookClient : IFacebookClient
+	{
+		private readonly HttpClient _httpClient;
+
+		public FacebookClient()
+		{
+			_httpClient = new HttpClient
+			{
+				BaseAddress = new Uri("https://graph.facebook.com/v18.0/")
+			};
+			_httpClient.DefaultRequestHeaders
+				.Accept
+				.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+		}
+
+		public async Task<T> GetAsync<T>(string accessToken, string endpoint, string args = null)
+		{
+			//endpoint example:accountId?fields=events
+			//args example: {updated_time,category,cover,description,end_time,event_times,id,is_canceled,is_draft,is_page_owned,name,place,start_time,type}
+			//full url example: https://graph.facebook.com/v18.0/yourAccountId?fields=events{category,updated_time,cover,description,end_time,event_times,id,is_canceled,is_draft,is_page_owned,name,place,start_time,type}&access_token=yourAccessToken"
+
+			var response = await _httpClient.GetAsync($"{endpoint}{args}&access_token={accessToken}");
```

Comment:
In epi, running this, I am getting an error - looks like there needs to be a ? between {endpoint} and {args}.

URL that threw error:
https://graph.facebook.com/v18.0/209783682775802/eventsfields=updated_time,category,cover,description,end_time,event_times,id,is_canceled,is_draft,is_page_owned,name,place,start_time,type&time_filter=upcoming&&access_token=EAACTONYWeFsBAP5gRZA7Lgw70DTcdIey5ep5n9eJp9HXGiuJplNJJgZCZCIk7YTYevokFEXYkJiiBZCs5IiXDQjSuy9TvdNeaqh8Rg3QoKdl4hWMS7rZBxtrZCkhdy5AJV7c3LQXHbfUUHhvSQ5nUFVWnTmTGx5MYZATc11w2L8eY2CBuQQI7iu

URL that worked:
https://graph.facebook.com/v18.0/209783682775802/events?fields=updated_time,category,cover,description,end_time,event_times,id,is_canceled,is_draft,is_page_owned,name,place,start_time,type&time_filter=upcoming&&access_token=EAACTONYWeFsBAP5gRZA7Lgw70DTcdIey5ep5n9eJp9HXGiuJplNJJgZCZCIk7YTYevokFEXYkJiiBZCs5IiXDQjSuy9TvdNeaqh8Rg3QoKdl4hWMS7rZBxtrZCkhdy5AJV7c3LQXHbfUUHhvSQ5nUFVWnTmTGx5MYZATc11w2L8eY2CBuQQI7iu

Having said that, the second URL "works" but doesn't return any data, not sure if that is the time_filter=upcoming.  If I remove that I get everything, but we don't necessarily want EVERYTHING anyway, just "new" stuff?